### ISSUE TITLE
Can't edit/change Templates Optional Settings

### ISSUE THREAD

**Issue by AlUser71** [akiko-pusu/redmine_issue_templates#322]
##  Summary
Templates Optional Settings can't be changed in version 1.0.0.

## Description
I check or uncheck one or all settings in the optional settings page.
When I try to save the chnaghed settings I get an error page "Page not found".

## Environment

System info: | Linux redmine-playground 4.15.0-76-generic #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
Environment:
  Redmine version                4.0.4.stable
  Ruby version                   2.5.1-p57 (2018-03-29) [x86_64-linux-gnu]
  Rails version                  5.2.3
  Environment                    production
  Database adapter               Mysql2
  Mailer queue                   ActiveJob::QueueAdapters::AsyncAdapter
  Mailer delivery                smtp
SCM:
  Subversion                     1.9.7
  Mercurial                      4.5.3
  Cvs                            1.12.13
  Bazaar                         2.8.0
  Git                            2.17.1
  Filesystem                     
  GitMirror                      2.17.1
Redmine plugins:
  additionals                    2.0.22-1
  easy_gantt                     1.12
  easy_mindmup                   1.0
  easy_wbs                       1.5
  redmine_agile                  1.5.2
  redmine_banner                 0.3.1
  redmine_checklists             3.1.17
  redmine_dmsf                   2.4.1
  redmine_favorite_projects      2.1.0
  redmine_git_mirror             0.8.0
  redmine_issue_templates        1.0.0
  redmine_questions              1.0.1
  redmine_resources              1.0.2
  redmine_smile_togglesidebar    1.0.7
  redmine_wiki_lists             0.0.9
  redmine_wysiwyg_editor         0.13.0
  redmine_xlsx_format_issue_exporter 0.1.6
  redmine_zenedit                1.0.3

## Visual Proof / Screenshot
![grafik](https://user-images.githubusercontent.com/60700838/76520294-efbdfe00-6462-11ea-966f-41821087f9ae.png)

![grafik](https://user-images.githubusercontent.com/60700838/76520328-fba9c000-6462-11ea-9121-fcb8338648e4.png)

[production.log](https://github.com/akiko-pusu/redmine_issue_templates/files/4324168/production.log)

## Expected Results
Settings can be changed and saved.

## Actual Results
Settings can't be changed, error page is displayed.

## Workaround
None.

**Comment by akiko-pusu** [comment]
Hi, @AlUser71 
Thanks for using my plugin.

I saw your attached log and found the problem. I really appreciate this reporting!
Here is a workawound:

```ruby
diff --git a/app/controllers/issue_templates_settings_controller.rb b/app/controllers/issue_templates_settings_controller.rb
index acc7ffa..c8f81a9 100644
--- a/app/controllers/issue_templates_settings_controller.rb
+++ b/app/controllers/issue_templates_settings_controller.rb
@@ -10,7 +10,7 @@ class IssueTemplatesSettingsController < ApplicationController
   end

   def edit
-    @issue_templates_setting = IssueTemplateSetting.find(@project.id)
+    @issue_templates_setting = IssueTemplateSetting.find_or_create(@project.id)
     return if params[:settings].blank?

     update_template_setting
```

Please replace issue_templates_settings_controller.rb line:13
from ``IssueTemplateSetting.find(@project.id)`` to ``IssueTemplateSetting.find_or_create(@project.id)``

Then, you can create a new issue template setting.
If you have some template settings created before using version0.3.x.

I'm going to release a fixed version soon.

**Comment by akiko-pusu** [comment]
Hi, I've just released v1.0.1 as a bugfix release.
Thanks for your valuable information.
